tests/go-validation/rancherrest/machine_pool.go

```diff
@@ -0,0 +1,58 @@
+package rancherrest
+
+import (
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	DOKind     = "DigitaloceanConfig"
+	DOPoolType = "rke-machine-config.cattle.io.digitaloceanconfig"
+)
+
+func NewMachinePoolConfig(generatedPoolName, kind, namespace, poolType, image, region, size string) *unstructured.Unstructured {
+	machineConfig := &unstructured.Unstructured{}
+	machineConfig.SetAPIVersion("rke-machine-config.cattle.io/v1")
+	machineConfig.SetKind(kind)
+	machineConfig.SetGenerateName(generatedPoolName)
+	machineConfig.SetNamespace(namespace)
+	machineConfig.Object["accessToken"] = ""
+	machineConfig.Object["image"] = image
+	machineConfig.Object["backups"] = false
+	machineConfig.Object["ipv6"] = false
+	machineConfig.Object["monitoring"] = false
+	machineConfig.Object["privateNetworking"] = false
+	machineConfig.Object["region"] = region
+	machineConfig.Object["size"] = size
+	machineConfig.Object["sshKeyContents"] = ""
+	machineConfig.Object["sshKeyFingerprint"] = ""
+	machineConfig.Object["sshPort"] = "22"
+	machineConfig.Object["sshUser"] = "root"
+	machineConfig.Object["tags"] = ""
+	machineConfig.Object["type"] = poolType
+	machineConfig.Object["userdata"] = ""
+	return machineConfig
+}
+
+func (p *PodConfigClient) CreateMachineConfigPool(machinePoolConfig *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	podResult, err := p.Namespace(machinePoolConfig.GetNamespace()).Create(p.Ctx, machinePoolConfig, metav1.CreateOptions{})
+	return podResult, err
+}
+
+func MachinePoolSetup(controlPlaneRole, etcdRole, workerRole bool, poolName string, quantity int32, machineConfig *unstructured.Unstructured) apisV1.RKEMachinePool {
```

Comment:
Should we add types for `controlPlaneRole` and `etcdRole` to be consistent?

Reply:
What do you mean Izaac? Have different setup methods?

Reply:
I'm just noting we're using types explicitly for `workerRole`, `poolName` and not for `controlPlaneRole` and `etcdRole`

Reply:
Ah I see what you're saying, sorry. This is a go convention thing where if you have multiple parameters with the same type you don't need to explicitly state the type for every one.

Reply:
Ah!! conventions, Ok i get it.